src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/NUnit/NUnitSkipCommandExecuteIntegration.cs

```diff
@@ -0,0 +1,40 @@
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.NUnit
+{
+    /// <summary>
+    /// NUnit.Framework.Internal.Commands.SkipCommand.Execute() calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        Assembly = "nunit.framework",
+        Type = "NUnit.Framework.Internal.Commands.SkipCommand",
+        Method = "Execute",
+        ReturnTypeName = "NUnit.Framework.Internal.TestResult",
+        ParametersTypesNames = new[] { "NUnit.Framework.Internal.TestExecutionContext" },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = IntegrationName)]
+    public class NUnitSkipCommandExecuteIntegration
+    {
+        private const string IntegrationName = "NUnit";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TContext">ExecutionContext type</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="executionContext">Execution context instance</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TContext>(TTarget instance, TContext executionContext)
```

Comment:
Is this missing an `OnMethodEnd` method? I expected there to be one like in the NUnitTestMethodCommandExecuteIntegration class

Reply:
No, the `OnMethodEnd` callback is optional, here because we are instrumenting `SkipCommand` (a skipped test) we can open and close the span in the same `OnMethodBegin` with the skip data.

Reply:
I added the `OnMethodEnd` in https://github.com/DataDog/dd-trace-dotnet/pull/1131/commits/7240eb751450f3782581f058795faadd2bd10ac3 to avoid confusions but I don't expect to ever been triggered with a scope instance in the state.

Reply:
Oh I see, in the common `CreateScope` code path, the scope would be closed, that makes sense. Thanks for the clarification!